libcontainer/system/linux.go

```diff
@@ -102,3 +105,29 @@ func GetSubreaper() (int, error) {
 
 	return int(i), nil
 }
+
+func ExecutableMemfd(comment string, flags int) (*os.File, error) {
+	// Try to use MFD_EXEC first. On pre-6.3 kernels we get -EINVAL for this
+	// flag. On post-6.3 kernels, with vm.memfd_noexec=1 this ensures we get an
+	// executable memfd. For vm.memfd_noexec=2 this is a bit more complicated.
+	// The original vm.memfd_noexec=2 implementation incorrectly silently
+	// allowed MFD_EXEC[1] -- this should be fixed in 6.6. On 6.6 and newer
+	// kernels, we will get -EACCES if we try to use MFD_EXEC with
+	// vm.memfd_noexec=2 (for 6.3-6.5, -EINVAL was the intended return value).
+	//
+	// The upshot is we only need to retry without MFD_EXEC on -EINVAL because
+	// it just so happens that passing MFD_EXEC bypasses vm.memfd_noexec=2 on
+	// kernels where -EINVAL is actually a security denial.
+	memfd, err := unix.MemfdCreate(comment, flags|unix.MFD_EXEC)
+	if errors.Is(err, unix.EINVAL) {
+		memfd, err = unix.MemfdCreate(comment, flags)
+	}
+	if err != nil {
+		if errors.Is(err, unix.EACCES) {
+			logrus.Infof("memfd_create(MFD_EXEC) failed, possibly due to vm.memfd_noexec=2 -- falling back to less secure O_TMPFILE")
```

Comment:
nit: `/Infof/Info/`

Also, `Info` is always shown, which might be a problem. Maybe change it to `Warn`?

Reply:
`Info` is a lower log level than `Warn`. We don't allow users to specify the log level from runc (maybe we should) but if `Info`-level messages are shown then so will `Warn`-level messages.

This also isn't really serious enough to be a "warning", but it's a hint that the admin probably should disable `vm.memfd_noexec=2`.